src/domains/ecommerce-common/02_security.tf

```diff
@@ -719,6 +719,34 @@ resource "azurerm_key_vault_secret" "service_management_opsgenie_webhook_token"
   }
 }
 
+resource "random_password" "ecommerce_payment_requests_primary_api_key_pass" {
+  length  = 32
+  special = false
+}
+
+resource "random_password" "ecommerce_payment_requests_secondary_api_key_pass" {
+  length  = 32
+  special = false
+}
+
+resource "azurerm_key_vault_secret" "ecommerce_payment_requests_primary_api_key" {
+  name         = "ecommerce-payment-requests-primary-api-key"
+  value        = random_password.ecommerce_payment_requests_primary_api_key_pass.result
+  key_vault_id = module.key_vault.id
+}
+
+resource "azurerm_key_vault_secret" "ecommerce_payment_requests_secondary_api_key" {
+  name         = "ecommerce-payment-requests-secondary-api-key"
+  value        = random_password.ecommerce_payment_requests_secondary_api_key_pass.result
+  key_vault_id = module.key_vault.id
+}
+
+resource "azurerm_key_vault_secret" "ecommerce_payment_requests_active_api_key" {
+  name         = "ecommerce-payment-requests-active-api-key"
+  value        = var.ecommerce_payment_requests_api_key_use_primary ? azurerm_key_vault_secret.ecommerce_payment_requests_primary_api_key.value : azurerm_key_vault_secret.ecommerce_payment_requests_secondary_api_key.value
```

Comment:
so to activate the secondary is it necessary to apply terraform with consequent restart of pods that read that secret?

Reply:
Thanks! however, could we populate the named value directly with one of the two secrets? this way, in case of rotation, only a single apply would be needed to inject it into the named value (currently, we need to do one apply in app and then another in common)